dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```diff
@@ -1114,6 +1115,39 @@ public void syncUserMetadataFromGitHubByUsername(User user, TokenDAO tokenDAO) {
         }
     }
 
+    /**
+     * DO NOT USE THIS FUNCTION ELSEWHERE.
+     * This function is for gathering topics for existing entries and only needs to be run once.
+     * @param entries A list of entries to set the topic for
+     * @return A list of entries that did not have their topics updated because of a failure in retrieving their topics from GitHub
+     */
+    public List<Entry> syncTopics(List<Entry> entries) {
+        List<Entry> entriesNotUpdatedWithTopic = new ArrayList<>();
+        GHRateLimit startRateLimit = getGhRateLimitQuietly();
+
+        for (Entry entry : entries) {
+            String repositoryId = getRepositoryId(entry);
+            GHRepository repository;
+            try {
+                repository = github.getRepository(repositoryId);
+            } catch (IOException e) {
+                LOG.info(String.format("Could not get topic from: %s", repositoryId), e);
+                entriesNotUpdatedWithTopic.add(entry);
+                continue;
+            }
+            String topic = repository.getDescription();
+            if (topic != null) { // Could be null if the repo doesn't have a description
+                topic = StringUtils.abbreviate(topic, MAX_TOPIC_LENGTH);
```

Comment:
Might be nice to have the GitHub-description-to-topic conversion code in a separate method that can be reused by any code that might import/update an entry.  That way, we don't repeat ourselves, and if we change the abbreviation algorithm, everything immediately works consistently everywhere.  Not sure this has to happen in this PR, but eventually, would be good...